module-new/src/components/checkboxList/checkboxList.component.tsx

```diff
@@ -0,0 +1,186 @@
+  bind?: IBindingProps<Id[]>;
+
+  /** The options to be shown in the input */
+  options: IArmstrongOption<
+    Id,
+    Omit<
+      ICheckboxProps<boolean>,
+      | 'bind'
+      | 'checked'
+      | 'disabled'
+      | 'label'
+      | 'validationErrorsClassName'
+      | 'validationErrorMessages'
+      | 'scrollValidationErrorsIntoView'
+      | 'ref'
+    >
+  >[];
+
+  /** CSS className property */
+  className?: string;
+
+  /** the current value of the radioInput */
+  value?: Id[];
+
+  /** Fired when the value changes */
+  onChange?: (newValue: Id[]) => void;
+
+  /** show an error state icon on the component (will be true automatically if validationErrorMessages are passed in or errors are in the binder) */
+  error?: boolean;
+
+  /** (Optional) A custom JSX.Element for the checked indicator. */
+  customIndicator?: JSX.Element;
+
+  /** which size variant to use */
+  displaySize?: DisplaySize;
+
+  /** Label for the whole radio group itself */
+  label?: string;
+
+  /** (Optional) Class name for the label component */
+  labelClassName?: string;
+
+  /** Indicates if field must be used to submit form */
+  required?: boolean;
+
+  /** wether input's value can be changed by user */
+  disabled?: boolean;
+
+  /** Symbol to use as the required indicator on the label, defaults to "*" */
+  requiredIndicator?: React.ReactNode;
+}
+
+/** Render a list of radio inputs which binds to a single string */
+export const CheckboxList = React.forwardRef<HTMLDivElement, ICheckboxListProps<ArmstrongId>>(
+  (
+    {
+      bind,
+      options,
+      className,
+      value,
+      errorIcon,
+      validationMode,
+      validationErrorMessages,
+      onChange,
+      customIndicator,
+      error,
+      displaySize,
+      label,
+      labelClassName,
+      required,
+      disabled,
+      scrollValidationErrorsIntoView,
+      requiredIndicator,
+      ...nativeProps
+    },
+    ref
+  ) => {
+    const globals = useArmstrongConfig({
+      validationMode,
+      requiredIndicator,
+      scrollValidationErrorsIntoView,
+      validationErrorIcon: errorIcon,
+    });
+
+    const [boundValue, setBoundValue, bindConfig] = useBindingState(bind, {
+      value,
+      validationErrorMessages,
+      validationErrorIcon: errorIcon,
+      validationMode,
+      onChange,
+    });
+
+    const onCheckedChange = React.useCallback(
+      (option: typeof options[0], newValue: boolean) => {
+        // Always mapping the bound value from the original array of options.
+        // It's a bit more cumbersome, but it means that the bound value ids will always be in the same order as the options, which is what you'd expect as a consuming app.
+        const newBoundValue = options.reduce((arr, op) => {
+          const isSelectedOption = op.id === option.id;
+          const isCurrentlyBound = boundValue?.includes(op.id);
+          const shouldAdd = isCurrentlyBound ? !isSelectedOption || newValue : isSelectedOption && newValue;
```

Comment:
might be tempted to break out the ternary fork values into `const`s just for readability in future / for other devs